kubernetes/resource_kubernetes_replication_controller.go

```diff
@@ -76,40 +76,23 @@ func resourceKubernetesReplicationController() *schema.Resource {
 
 func replicationControllerTemplateFieldSpec() map[string]*schema.Schema {
 	metadata := namespacedMetadataSchemaIsTemplate("replication controller's template", true, true)
-	// TODO: make this required once the legacy fields are removed
-	metadata.Computed = true
-	metadata.Required = false
-	metadata.Optional = true
+	metadata.Required = true
 
 	templateFields := map[string]*schema.Schema{
 		"metadata": metadata,
 		"spec": {
 			Type:        schema.TypeList,
 			Description: "Spec of the pods managed by the replication controller",
-			Optional:    true, // TODO: make this required once the legacy fields are removed
-			Computed:    true,
+			Required:    true,
 			MaxItems:    1,
 			Elem: &schema.Resource{
-				Schema: podSpecFields(false, false, true),
+				Schema: podSpecFields(false, true),
```

Comment:
Now that I'm reviewing this again, I see it sets `isComputed` to true even though this isn't a computed field. I know I tested the end result of this pretty thoroughly, but I wonder if there were any unforeseen impacts of having `isComputed=true`,

Reply:
That's far away in my mind now, but re-reading the change https://github.com/hashicorp/terraform-provider-kubernetes/pull/193/files#diff-5f4831c5771d0f4db1c84186dd377b99c788922fe9dfd33171a92de100426f15R7, I believe I did pass `true` to `isComputed` here to allow for a transition from the deprecated fields to the new ones in existing configurations. Not doing this would have triggered use diffs on those new fields, I think.